src/cm_common/cm_utils.c

```diff
@@ -520,12 +520,17 @@ read_server_status_output (T_SERVER_STATUS_RESULT * res, char *out_file)
       if (num_info > num_alloc)
 	{
 	  num_alloc += 5;
-	  info = (T_SERVER_STATUS_INFO *) realloc (info, sizeof (T_SERVER_STATUS_INFO) * num_alloc);
-	  if (info == NULL)
+	  T_SERVER_STATUS_INFO *const new_info
+	    = (T_SERVER_STATUS_INFO *) realloc (info, sizeof (T_SERVER_STATUS_INFO) * num_alloc);
+	  if (new_info == NULL)
 	    {
 	      fclose (fp);
```

Comment:
If new_info is NULL, info also needs to free memory.

Reply:
Your comment is accepted. Thank you.